otherlibs/unix/open.c

```diff
@@ -20,20 +20,13 @@
 #include <caml/signals.h>
 #include "unixsupport.h"
 #include <string.h>
-#ifdef HAS_UNISTD
 #include <unistd.h>
-#endif
 #include <fcntl.h>
 
-#ifndef O_NONBLOCK
-#define O_NONBLOCK O_NDELAY
-#endif
+/* O_DSYNC and O_RSYNC are part of POSIX Synchronized IO option */
```

Comment:
Should we be detecting these with a feature-test macro rather than `ifndef O_DSYNC`? I'm not sure whether POSIX guarantees that `O_DSYNC` is exposed as a macro rather than an enum, so I don't fully trust the ifdef test.

Reply:
POSIX does guarantee that those are macros, yes (with its "this must be suitable for use in `#if` language).

The issue here is FreeBSD (and possibly macOS + OmniOS, but I just double-checked on FreeBSD - I originally saw it on all three of those with `O_RSYNC` when running an earlier version through precheck). FreeBSD 13 has `O_DSYNC` but no FreeBSD has `O_RSYNC`. Consequently FreeBSD defines `_POSIX_SYNCHRONIZED_IO` as `-1` which also, in passing, reveals a bug-ish in `mmap.c` which is indirectly fixed by this PR.

Reply:
> POSIX does guarantee that those are macros, yes (with its "this must be suitable for use in #if language).

That is good to know, thanks!